lib/rules/space-infix-ops.js

```diff
@@ -4,6 +4,98 @@
  */
 "use strict";
 
+/**
+ * Returns the first token which violates the rule
+ * @param {SourceCode} sourceCode The source code object to get the operator
+ * @param {ASTNode} left - The left node of the main node
+ * @param {ASTNode} right - The right node of the main node
+ * @param {string} op - The operator of the main node
+ * @returns {Object} The violator token or null
+ * @private
+ */
+function getFirstNonSpacedToken({ sourceCode, left, right, op }) {
+    const operator = sourceCode.getFirstTokenBetween(left, right, token => token.value === op);
+    const prev = sourceCode.getTokenBefore(operator);
+    const next = sourceCode.getTokenAfter(operator);
+
+    if (!sourceCode.isSpaceBetweenTokens(prev, operator) || !sourceCode.isSpaceBetweenTokens(operator, next)) {
+        return operator;
+    }
+
+    return null;
+}
+
+/**
+ * Reports an AST node as a rule violation
+ * @param {SourceCode} sourceCode - The source code object to get the tokens
+ * @param {ASTNode} mainNode - The node to report
+ * @param {Object} culpritToken - The token which has a problem
+ * @returns {MessageDescriptor} descriptor A descriptor for the report from a rule.
+ * @private
+ */
+function getDescriptor({ sourceCode, mainNode, culpritToken }) {
+    return {
+        node: mainNode,
+        loc: culpritToken.loc.start,
+        message: "Operator '{{operator}}' must be spaced.",
+        data: {
+            operator: culpritToken.value
+        },
+        fix(fixer) {
+            const previousToken = sourceCode.getTokenBefore(culpritToken);
+            const afterToken = sourceCode.getTokenAfter(culpritToken);
+            let fixString = "";
+
+            if (culpritToken.range[0] - previousToken.range[1] === 0) {
+                fixString = " ";
+            }
+
+            fixString += culpritToken.value;
+
+            if (afterToken.range[0] - culpritToken.range[1] === 0) {
+                fixString += " ";
+            }
+
+            return fixer.replaceText(culpritToken, fixString);
+        }
+    };
+}
+
+/**
+ * Check if the node is binary then report
+ * @param {SourceCode} sourceCode - sourceCode object of the file
+ * @param {ASTNode} node node to evaluate
+ * @param {boolean} int32Hint is int32Hint or not
+ * @returns {MessageDescriptor | null} descriptor A descriptor for the report from a rule.
+ * @public
+ */
+function checkBinary({ sourceCode, node, int32Hint }) {
+    const leftNode = (node.left.typeAnnotation) ? node.left.typeAnnotation : node.left;
```

Comment:
is there a way not to check the `node.left.typeAnnotation` existence; better not to do this, as it is typescript-specific.